functions/index.ts

```diff
@@ -0,0 +1,168 @@
+        HttpResponse({statusCode: 200, body: JSON.stringify(game)});
+      } else {
+        throw Error;
+     }
+    } catch (err) {
+      throw HttpResponse({statusCode: 404, body: `Unable to find game.`});
+    }
+  }
+  
+  async function listGames() {
+    let params = {
+      TableName: table,
+      Select: "ALL_ATTRIBUTES"
+    };
+    
+    try {
+      let response = await docClient.scan(params).promise();
+      let gameList : Game[] = [];
+      
+      if (response.Items) {
+        response.Items.forEach((game: Game) => {
+          gameList.push(new Game(game.gameName, game.yearReleased, game.genre, game.console, game.developer));
+        })
+        HttpResponse({statusCode: 200, body: JSON.stringify(gameList)})
+      } else {
+        throw Error;
+     }
+    } catch (err) {
+      throw HttpResponse({statusCode: 404, body: `Game list could not be found.`});
+    }
+  }
+
+  async function createGame(game: Game) {
+    let params = {
+      TableName: table,
+      Item: {
+        gameName: game.gameName,
+        genre: game.genre,
+        yearReleased: game.yearReleased,
+        developer: game.developer,
+        console: game.console
+      },
+      ConditionExpression: 'attribute_not_exists(gameName)'
+    }
+
+    try {
+      let response = await docClient.put(params).promise();
+      if (Object.keys(response).length == 0) {
+        HttpResponse({statusCode: 201, body: JSON.stringify(game)});
+      } else {
+        throw Error;
+     }
+    } catch(err) {
+      throw HttpResponse({statusCode: 403, body: `Unable to create game.`});
+    }
+  }
+
+  async function modifyGame(game: Game) {
+    let updateExpression: String[] = [];
+    let expressionAttributeNames = {} as any;
+    let expressionAttributeValues = {} as any;
+    let body = JSON.parse(event.body);
+    
+    //Generate dynammic update expression based on allowed parameters
+    for (let parameter in body) {
+      if (allowedRequestParameters.includes(parameter)) {
+        updateExpression.push(`#${parameter} = :${parameter}`);
+        expressionAttributeNames['#'+parameter] = parameter ;
+        expressionAttributeValues[':'+parameter]=`${body[parameter]}`;  
+      }
+    }
+    
+    let params = {
+      TableName: table,
+      Key: {
+        gameName: body.gameName
+      },
+      UpdateExpression: `SET ${updateExpression.join(",")}`,
+      ExpressionAttributeNames: expressionAttributeNames,
+      ExpressionAttributeValues: expressionAttributeValues,
+      ConditionExpression: 'attribute_exists(gameName)',
+      ReturnValues: 'ALL_NEW'
+    };
+    
+    try {
+      let response = await docClient.update(params).promise();
+      if (response.Attributes) {
+        HttpResponse({statusCode: 200, body: JSON.stringify(game)});
+      } else {
+        throw Error;
+      }
+    } catch (err) {
+      throw HttpResponse({statusCode: 403, body: `Unable to modify game.`});
```

Comment:
403 indicates it was a problem with their request, but you don't validate that here. What if the error was just a general DynamoDB throttling error?